### PR TITLE
Add `validate` parameter to `torch.multinomial` to skip input validation

### PR DESCRIPTION
**Pull request by Nik-Reddy**
## Summary

Adds a `validate` keyword argument (default `True`) to `torch.multinomial`. When `validate=False`, the 10 GPU validation kernels (`aminmax`, `sum`, `assert_async`, etc.) on the fast path (`!with_replacement || n_sample == 1`) are skipped entirely.

Fixes #177127

## Motivation

As profiled in the issue, `torch.multinomial` spends **~66% of GPU time** (~107 µs out of ~161 µs on an RTX 3090 with V=128,000) on input validation — checking for negative values, NaN, Inf, and zero-sum distributions. This validation is unnecessary when the caller knows the input is valid, e.g. when probabilities come directly from `softmax()`, which guarantees values in [0, 1], no NaN/Inf, and sum ~ 1.0.

This matters for **LLM inference**, where `torch.multinomial` is called on every decode step with softmax output. Skipping validation yields an estimated **~3x speedup** for this hot-path operation.

## Changes

| File | Change |
|------|--------|
| `native_functions.yaml` | Add `bool validate=True` as keyword-only arg to both `multinomial` and `multinomial.out` |
| `Distributions.cpp` | Guard the fast-path validation block with `if (validate)` |
| `Distributions.mm` (MPS) | Same guard for the MPS backend |
| `derivatives.yaml` | Update signature |
| `_meta_registrations.py` | Accept new kwarg in meta function |
| `overrides.py` | Update lambda signature |
| `_torch_docs.py` | Document the `validate` kwarg with usage guidance |
| `test_torch.py` | Add `test_multinomial_validate_false` covering 1D/2D, replacement/no-replacement, method variant, and softmax input |
| `common_methods_invocations.py` | Add `validate=False` sample inputs to OpInfo |

## Usage

```python
# Default behavior unchanged (validation on):
torch.multinomial(probs, num_samples=1)

# Skip validation when input is known-valid (e.g. from softmax):
torch.multinomial(probs, num_samples=1, validate=False)

# Method variant also supported:
probs.multinomial(num_samples=1, validate=False)
```

## Design Decisions

- **Keyword-only**: `validate` is keyword-only to prevent positional ambiguity with existing args
- **Default `True`**: Full backward compatibility — existing code is unaffected
- **Name choice**: `validate` follows the pattern used by other PyTorch APIs (e.g. `torch.nn.utils.clip_grad_norm_` uses similar optional validation flags)
- **Both CPU and GPU**: Validation is guarded on both `Distributions.cpp` (CPU/CUDA) and `Distributions.mm` (MPS)

## Testing

- Added `test_multinomial_validate_false` with coverage for all code paths (1D, 2D, with/without replacement, method variant, softmax input)
- Added `validate=False` sample inputs to OpInfo for broader test matrix coverage
- Existing tests pass unchanged (`validate=True` is the default)

cc @jerryzh168 @ptrblck @msaroufim @eqy @tinglvv @nWEIdia

### CODE REVIEW

**Comment by Nik-Reddy** [comment]
Pushed a fix for the Mac MPS failures — the VmapMode dispatch registration in `VmapModeRegistrations.cpp` was still using the old 4-arg signature for `aten::multinomial` (missing the new `validate` bool). FuncTorchVmapMode already had the 5-arg version from my earlier commit, so the mismatch was crashing at import time.

Regarding the FC-window warning on `native_functions.yaml` — I understand the policy. Happy to split this into two PRs if that's preferred: one for the C++ side (native_functions + kernel impls) and a follow-up for the Python frontend changes after the FC window.

**Comment by Nik-Reddy** [comment]
Hi @malfet @albanD, gentle ping on this. The change is opt-in (`validate=False`, default stays True so current behavior is unchanged) and CI is green. Could one of you take a look when you get a chance, or point me to a better owner for `torch.multinomial`?
